### PR TITLE
Add sodaq autonomo

### PR DESCRIPTION
**Pull request by keestux**
This adds support for a new board, the SODAQ Autonomo. It is loosly based on Arduino Zero, but with a SAMD21J18. Other features include:
- 16Mbit dataflash
- SDcard
- bee compatible slot
- Solar charger controller

### CODE REVIEW

**Comment by OlegHahm** [comment]
Is this PR based on another one? If not, it probably should be split into two PRs.

File: boards/sodaq-autonomo/include/board.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+#define LED_PORT            PORT->Group[PA]
+#define LED0_MASK           (1 << 18)
+
+#define LED0_ON             (LED_PORT.OUTSET.reg = LED0_MASK)
+#define LED0_OFF            (LED_PORT.OUTCLR.reg = LED0_MASK)
+#define LED0_TOGGLE         (LED_PORT.OUTTGL.reg = LED0_MASK)
+/** @} */
+
+#if 0
```
Comment (aabadie) [inline boards/sodaq-autonomo/include/board.h]: Just asking: why excluding this block of code ?

File: cpu/sam21_common/include/cmsis/samd21/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+the variant, for example `__SAMD21J18A__`.  This define must be set in the
+`Makefile.include` of the board.
+
+Be aware that if you want to make changes to any file in this tree that the
+changes will be lost when a new ASF release is going to be used.
+
+## Trailing White Space
+
+Because of the whitespace check (dist/tools/whitespacecheck/check.sh) all
+the trailing white space had to be removed.  Please take this into account
```
Comment (aabadie) [inline cpu/sam21_common/include/cmsis/samd21/README.md]: spaces

File: boards/sodaq-autonomo/Makefile.dep [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Not sure if this is needed
```
Comment (aabadie) [inline boards/sodaq-autonomo/Makefile.dep]: This comment can be removed as you added `gpio_params.h` where you configured a saul entry for the on-board LED. So leaving the `saul_gpio` can be useful.

File: boards/sodaq-autonomo/include/board.h (hunk 1 above)
Comment (keestux) [inline boards/sodaq-autonomo/include/board.h]: Good point. I'll remove it.

File: boards/sodaq-autonomo/Makefile.dep (hunk 3 above)
Comment (keestux) [inline boards/sodaq-autonomo/Makefile.dep]: OK, comment dropped

Thanks for looking at the PR

**Comment by keestux** [comment]
@aabadie Anything else?

File: cpu/sam21_common/include/cmsis/samd21/README.md (hunk 2 above)
Comment (aabadie) [inline cpu/sam21_common/include/cmsis/samd21/README.md]: Ok, got it.

**Comment by aabadie** [comment]
@keestux, in case you missed it, Murdock reported errors with `unittests`.

**Comment by keestux** [comment]
Is Murdock upset? The "Details" link gives a 404

**Comment by miri64** [comment]
@keestux it was restarted and since it is not retaining its results but in memory it forgot about your PR. Nothing another trigger can't fix ;-).

**Comment by keestux** [comment]
@miri64 Thanks

**Comment by aabadie** [comment]
ACK when a bit of documentation about the board is added on the [supported platform wiki page](https://github.com/RIOT-OS/RIOT/wiki/RIOT-Platforms). You can take inspiration on [samr21-xpro page](https://github.com/RIOT-OS/RIOT/wiki/Board%3A-Samr21-xpro).

**Comment by keestux** [comment]
Content added to the wiki. Basic info is present.

**Comment by aabadie** [comment]
Thanks for the wiki page @keestux. Merging now.

**Pull request merged**